kratos/geometries/geometry.h

```diff
@@ -410,10 +423,93 @@ class Geometry : public PointerVector<TPointType>
     //     return p_clone;
     // }
 
-    //lumping factors for the calculation of the lumped mass matrix
-    virtual Vector& LumpingFactors( Vector& rResult )  const
-    {
-        KRATOS_ERROR << "Called the virtual function for LumpingFactors " << *this << std::endl;
+    /**
+     * @brief Lumping factors for the calculation of the lumped mass matrix
+     * @param rResult Vector containing the lumping factors
+     * @param LumpingMethod The lumping method considered. The three methods available are:
+     *      - The row sum method
+     *      - Diagonal scaling
+     *      - Evaluation of M using a quadrature involving only the nodal points and thus automatically yielding a diagonal matrix for standard element shape function
+     */
+    virtual Vector& LumpingFactors(
+        Vector& rResult,
+        const LumpingMethods LumpingMethod = LumpingMethods::ROW_SUM
+        )  const
+    {
+        const SizeType number_of_nodes = this->size();
+        const SizeType local_space_dimension = this->LocalSpaceDimension();
+
+        // Clear lumping factors
+        if (rResult.size() != number_of_nodes)
+            rResult.resize(number_of_nodes, false);
+        rResult = ZeroVector(number_of_nodes);
+
+        if (LumpingMethod == LumpingMethods::ROW_SUM) {
+            IntegrationMethod integration_method = GetDefaultIntegrationMethod();
+            integration_method = static_cast<IntegrationMethod>(static_cast<int>(integration_method) + 1);
+            const GeometryType::IntegrationPointsArrayType& integration_points = this->IntegrationPoints( integration_method );
+            const Matrix& Ncontainer = this->ShapeFunctionsValues(integration_method);
+
+            for ( IndexType point_number = 0; point_number < integration_points.size(); ++point_number ) {
+                const double detJ = DeterminantOfJacobian(point_number, integration_method);
```

Comment:
I think it would be faster to use the version that computes all the determinants at once

Reply:
OK